src/Hl7.Fhir.Patch/ElementNodePatchExtensions.cs

```diff
@@ -0,0 +1,113 @@
+/* 
+ * Copyright (c) 2020, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://github.com/FirelyTeam/fhir-net-api/blob/master/LICENSE
+ */
+
+using System;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Patch.Adapters;
+
+namespace Hl7.Fhir.Patch
+{
+    public static class ElementNodePatchExtensions
```

Comment:
My thinking here was:

- If a user wants to chain multiple patch documents, the `ElementNode` overload will prevent making unnecessary copies
- If a user already has a mutable object, it makes sense to allow mutating it, rather than making a copy

But I agree, if there is no need to support the 2 use cases above, I will make the `ElementNode` overloads internal.